mail_notification_custom_subject/models/mail_message_custom_subject.py

```diff
@@ -0,0 +1,158 @@
+from werkzeug import urls
+
+from odoo import _, api, fields, models, tools
+from odoo.exceptions import UserError
+from odoo.tools import pycompat
+
+_logger = logging.getLogger(__name__)
+
+try:
+    # We use a jinja2 sandboxed environment to render mako templates.
+    # Note that the rendering does not cover all the mako syntax, in particular
+    # arbitrary Python statements are not accepted, and not all expressions are
+    # allowed: only "public" attributes (not starting with '_') of objects may
+    # be accessed.
+    # This is done on purpose: it prevents incidental or malicious execution of
+    # Python code that may break the security of the server.
+    from jinja2.sandbox import SandboxedEnvironment
+    mako_template_env = SandboxedEnvironment(
+        block_start_string="<%",
+        block_end_string="%>",
+        variable_start_string="${",
+        variable_end_string="}",
+        comment_start_string="<%doc>",
+        comment_end_string="</%doc>",
+        line_statement_prefix="%",
+        line_comment_prefix="##",
+        trim_blocks=True,               # do not output newline after blocks
+        autoescape=True,                # XML/HTML automatic escaping
+    )
+    mako_template_env.globals.update({
+        'str': str,
+        'quote': urls.url_quote,
+        'urlencode': urls.url_encode,
+        'datetime': datetime,
+        'len': len,
+        'abs': abs,
+        'min': min,
+        'max': max,
+        'sum': sum,
+        'filter': filter,
+        'reduce': functools.reduce,
+        'map': map,
+        'round': round,
+
+        # dateutil.relativedelta is an old-style class and cannot be directly
+        # instanciated wihtin a jinja2 expression, so a lambda "proxy" is
+        # is needed, apparently.
+        'relativedelta': lambda *a, **kw : relativedelta.relativedelta(*a, **kw),
+    })
+    mako_safe_template_env = copy.copy(mako_template_env)
+    mako_safe_template_env.autoescape = False
+except ImportError:
+    _logger.warning("jinja2 not available, templating features will not work!")
+
+
+class MailMessageCustomSubject(models.Model):
+    _name = "mail.message.custom.subject"
+    _description = 'Mail Message Custom Subject'
+
+    name = fields.Char(string="Template Name")
+    model_id = fields.Many2one(
+        comodel_name="ir.model", string="Models", required=True,
+        help="Model(s) where this template applies",
+    )
+    subtype_ids = fields.Many2many(
+        comodel_name='mail.message.subtype', string='Subtype', required=True,
+        help="Message subtypes followed, meaning subtypes that will be pushed onto the user's Wall")
+    subject_template = fields.Char(
+        string='Subject Template', required=True,
+        help="Subject (placeholders may be used here)"
+    )
+    replace = fields.Boolean(
+        string="Replace", default=True,
+        help="Whether to replace or append to other templates that apply to a"
+        " given context"
+    )
```

Comment:
Append allways after right?
Maybe need to change `replace` field to `position` (select): replace/append_after/append_before

What do you think?

Reply:
That may be an interesting option. @pedrobaeza WDYT?

Reply:
OK